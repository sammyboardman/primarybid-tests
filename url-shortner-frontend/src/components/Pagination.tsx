import React, { useState } from "react";
import { MdArrowBackIos, MdArrowForwardIos } from "react-icons/md";
import { IPaginationProps } from "../interfaces";

const Pagination: React.FC<IPaginationProps> = (props) => {
  const pageNumbers = [];
  const [selectedPage, setSelectedPage] = useState<number>(1);

  for (let i = 1; i <= props.pageCount; i++) {
    pageNumbers.push(i);
  }

  const nextPage = () => {
    if (props.page < props.pageCount) {
      const pageNumber = props.page + 1;
      props.setPage(pageNumber);
      setSelectedPage(pageNumber);
    }
  };

  const prevPage = () => {
    if (props.page > 1) {
      const pageNumber = props.page - 1;
      props.setPage(pageNumber);
      setSelectedPage(pageNumber);
    }
  };

  const goToPage = (num: number) => {
    props.setPage(num);
    setSelectedPage(num);
  };

  return (
    <nav>
      <ul className="pagination">
        <li className="page-item" onClick={prevPage}>
          <MdArrowBackIos title="arrow-back-id" />
        </li>
        {pageNumbers.map((number) => (
          <li
            key={number}
            className={
              selectedPage === number
                ? "active-page-item"
                : "page-item"
            }
            onClick={() => goToPage(number)}
          >
            {number}
          </li>
        ))}
        <li className="page-item" onClick={nextPage}>
          <MdArrowForwardIos title="arrow-next-id" />
        </li>
      </ul>
    </nav>
  );
};

export default Pagination;
