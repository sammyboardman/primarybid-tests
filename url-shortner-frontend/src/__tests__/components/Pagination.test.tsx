import { render, fireEvent } from "@testing-library/react";
import Pagination from "../../components/Pagination";

// Mock the setPage function
const mockSetPage = jest.fn();

const defaultProps = {
  pageCount: 5,
  page: 1,
  setPage: mockSetPage,
};

test("renders pagination correctly", () => {
  const { container, getAllByRole } = render(<Pagination {...defaultProps} />);

  expect(container).toMatchSnapshot();

  const paginationItems = getAllByRole("listitem");
  expect(paginationItems).toHaveLength(7);
});

test("clicking next button calls nextPage function", () => {
  const { getByTitle } = render(<Pagination {...defaultProps} />);

  const nextButton = getByTitle("arrow-next-id");
  fireEvent.click(nextButton);

  expect(mockSetPage).toHaveBeenCalledWith(2);
});

test("clicking previous button calls prevPage function", () => {
  const {  getByTitle } = render(<Pagination {...defaultProps} page={2} />);

  const prevButton = getByTitle("arrow-back-id");
  fireEvent.click(prevButton);

  expect(mockSetPage).toHaveBeenCalledWith(1);
});

test("clicking page number calls goToPage function", () => {
  const { getByText } = render(<Pagination {...defaultProps} />);

  const pageNumber3 = getByText("3");
  fireEvent.click(pageNumber3);

  expect(mockSetPage).toHaveBeenCalledWith(3);
});
