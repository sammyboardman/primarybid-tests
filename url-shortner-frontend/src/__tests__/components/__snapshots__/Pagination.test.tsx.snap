// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders pagination correctly 1`] = `
<div>
  <nav>
    <ul
      class="pagination"
    >
      <li
        class="page-item"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            arrow-back-id
          </title>
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M11.67 3.87L9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
          />
        </svg>
      </li>
      <li
        class="active-page-item"
      >
        1
      </li>
      <li
        class="page-item"
      >
        2
      </li>
      <li
        class="page-item"
      >
        3
      </li>
      <li
        class="page-item"
      >
        4
      </li>
      <li
        class="page-item"
      >
        5
      </li>
      <li
        class="page-item"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            arrow-next-id
          </title>
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M6.23 20.23L8 22l10-10L8 2 6.23 3.77 14.46 12z"
          />
        </svg>
      </li>
    </ul>
  </nav>
</div>
`;
